src/multibody/joint/joint-base.hpp

```diff
@@ -243,15 +249,80 @@ namespace se3
                   const bool update_I = false) const
     { return static_cast<const JointModel*>(this)->calc_aba(data, I, update_I); }
 
+
+    /**
+     * @brief      Integrate joint's configuration for a constant derivative during one unit time
+     *
+     * @param[in]  q     initatial configuration  (size jmodel.nq)
+     * @param[in]  v     joint velocity (size jmodel.nv)
+     *
+     * @return     The configuration integrated
+     */
+    ConfigVector_t integrate(const Eigen::VectorXd & q,const Eigen::VectorXd & v) const
+    { return derived().integrate_impl(q, v); } 
+
+
+    /**
+     * @brief      Interpolation between two joint's configurations
+     *
+     * @param[in]  q0    Initial configuration to interpolate
+     * @param[in]  q1    Final configuration to interpolate
+     * @param[in]  u     u in [0;1] position along the interpolation.
+     *
+     * @return     The interpolated configuration (q0 if u = 0, q1 if u = 1)
+     */
+    ConfigVector_t interpolate(const Eigen::VectorXd & q0,const Eigen::VectorXd & q1, double u) const
+    { return derived().interpolate_impl(q0, q1, u); }
+
+
+    /**
+     * @brief      Generate a random joint configuration, normalizing quaternions when necessary.
+     *
+     * \warning    Do not take into account the joint limits. To shoot a configuration uniformingly
+     *             depending on joint limits, see uniformySample
+     *
+     * @return     The joint configuration
+     */
+    ConfigVector_t random() const
+    { return derived().random_impl(); } 
+
+    /**
+     * @brief      Uniformly sample the configuration space of the joint
+     *
+     *
+     * @return     The joint configuration
+     */
+    ConfigVector_t uniformlySample(const ConfigVector_t & lower_pos_limit, const ConfigVector_t & upper_pos_limit) const
+    { return derived().uniformlySample_impl(lower_pos_limit, upper_pos_limit); } 
+
+    
+    /**
+     * @brief      the constant derivative that must be integrated during one unit time to go from q0 to q1
```

Comment:
Can you **change constant** derivative into **tangent vector**? This sounds much more correct from a mathematical view point.